pypeit/core/coadd.py

```diff
@@ -198,69 +198,97 @@ def unpack_spec(spectra, all_wave=False):
     return fluxes, sigs, wave
 
 
-def sn_weight(spectra, smask, debug=False):
-    """ Calculate the S/N of each input spectrum and
-    create an array of weights by which to weight the
-    spectra by in coadding.
+def sn_weights(flux, sig, mask, wave, dv_smooth=10000.0, const_weights=False, debug=False, verbose=False):
+    """ Calculate the S/N of each input spectrum and create an array of (S/N)^2 weights to be used
+    for coadding.
 
     Parameters
     ----------
-    spectra : XSpectrum1D
+    fluxes: float ndarray, shape = (nexp, nspec)
+        Stack of (nexp, nspec) spectra where nexp = number of exposures, and nspec is the length of the spectrum.
+    sigs: float ndarray, shape = (nexp, nspec)
+        1-sigam noise vectors for the spectra
+    mask_stack: bool ndarray, shape = (nexp, nspec)
+        Mask for stack of spectra. True=Good, False=Bad.
+    wave: flota ndarray, shape = (nspec,) or (nexp, nspec)
+        Reference wavelength grid for all the spectra. If wave is a 1d array the routine will assume
+        that all spectra are on the same wavelength grid. If wave is a 2-d array, it will use the individual
+
```

Comment:
Add a note on dv_smooth.
Is 10,000 km/s a bit large?

Reply:
This is the number that we always used in xidl. The idea is for the weights to be very smooth.